### ISSUE TITLE
Editor support for vscode

### ISSUE THREAD

**Issue by Victorious3** [neogeny/TatSu#107]
I thought I should post this here since the repo contains grammars for other editors, and it could be useful to someone.

I've written a vscode addon a while ago to support syntax highlighting and eventually autocompletion for symbols: https://github.com/Victorious3/vscode-TatSu

It is quite involved, but sadly unfinished since I discovered a glaring issue in vscode that makes writing it a complete pain. 

Waiting for this PR: https://github.com/Microsoft/vscode-languageserver-node/pull/367 
And this: https://github.com/Microsoft/vscode/issues/580

Apart from that, writing typescript is only marginally less annoying than writing Javascript so my motivation to enhance it is limited.

![Screenshot (43)](https://user-images.githubusercontent.com/3749322/56498716-51037880-6503-11e9-92dc-00eeb0b2cbb8.png)

It's not on the Marketplace yet since its performance characteristics are a bit... questionable (due to the issue mentioned above, it parses the entire file on every edit) and its lacking support for goto symbol.

**Comment by apalala** [comment]
Please add a pull request for this @Victorious3, so it gets included in the next release. It's the best way to gather help for finishing the work.

**Comment by Victorious3** [comment]
@apalala Aight, I can do that. Would you publish the addon when its done?

I think I can also include my updated tmlanguage file since it contains some improvements over [this one](https://github.com/neogeny/TatSu/blob/master/etc/sublime/tatsu.YAML-tmLanguage).

**Comment by apalala** [comment]
We'd publish it right-away, with a _caveat_.

**Comment by Victorious3** [comment]
I'm not sure how to include this into the project structure...
I would use a seperate repo, with a submodule perhaps? 
Feel free to pull it from my repo and integrate it as you see fit.

**Comment by apalala** [comment]
Oh! It's not a single program file, but a complete VS project. We should mention it in the documentation, then.

**Comment by FezVrasta** [comment]
@Victorious3 does your proof of concept include a working language parser?

I'm trying to generate an abstract syntax tree from a TatSu EBNF on the client-side (JavaScript).

**Comment by Victorious3** [comment]
@FezVrasta No, I'm using the textmate grammar, its not a complete parser, only enough to provide syntax highlighting

**Comment by apalala** [comment]
I've never used VSCode. This issue requires a pull request. Closing.
